src/Discovery/DiscoveryActionBar.tsx

```diff
@@ -27,8 +29,117 @@ interface Props {
   filtersVisible: boolean;
   setFiltersVisible: (boolean) => void;
   user: User,
+  downloadInProgress: boolean,
+  setDownloadInProgress: (boolean) => void,
+  downloadStatusMessage: {
+    title: string,
+    message: string,
+    active: boolean
+  },
+  setDownloadStatusMessage: (Object) => void;
 }
 
+const handleDownloadZipClick = async (
+  selectedResources: any[],
+  setDownloadInProgress: (boolean) => void,
+  setDownloadStatusMessage: (object) => void,
+  manifestFieldName: string,
+  maxDownloadSizeBytes: number,
+) => {
+  const studyIDs = selectedResources.map((study) => study.project_number);
+  let downloadSize = 0;
+  selectedResources.forEach(
+    (study) => {
+      study[manifestFieldName].forEach(
+        (file) => {
+          downloadSize += file.file_size;
+        },
+      );
+    },
+  );
+
+  if (downloadSize > maxDownloadSizeBytes) {
+    const maxSizeMB = (maxDownloadSizeBytes / 1000000).toFixed(1);
+    const downloadSizeMb = (downloadSize / 1000000).toFixed(1);
```

Comment:
FYI: for these 2, might be better just import the `humanFileSize` function from [https://github.com/uc-cdis/data-portal/blob/44921cc423704d8edb8da2a1467200c234d5625b/src/utils.js#L4